driver-core/src/main/com/mongodb/internal/Exceptions.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal;
+
+import com.mongodb.MongoCommandException;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonNumber;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.mongodb.internal.VisibleForTesting.AccessModifier.PRIVATE;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public final class Exceptions {
+    public static final class MongoCommandExceptions {
+        public static int extractErrorCode(final BsonDocument response) {
+            return extractErrorCodeAsBson(response).intValue();
+        }
+
+        public static String extractErrorCodeName(final BsonDocument response) {
+            return extractErrorCodeNameAsBson(response).getValue();
+        }
+
+        public static BsonArray extractErrorLabelsAsBson(final BsonDocument response) {
```

Comment:
Optional: move this down near the other `asBson`-suffixed private methods.  It caused some cognitive dissonance during review to see it here.

Reply:
Unlike those methods, this one is `public`, and I organized the methods such that `private` ones are ordered after the `public` methods. I reordered the methods.